src/Type/EntityQuery/EntityQueryHasAccessCheckDynamicReturnTypeExtension.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace mglaman\PHPStanDrupal\Type\EntityQuery;
+
+use Drupal\Core\Entity\Query\QueryInterface;
+use PhpParser\Node\Expr\MethodCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\MethodReflection;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\Type\DynamicMethodReturnTypeExtension;
+use PHPStan\Type\Type;
+
+class EntityQueryHasAccessCheckDynamicReturnTypeExtension implements DynamicMethodReturnTypeExtension
```

Comment:
`EntityQueryAccessCheckDynamicReturnTypeExtension` rename since it's for `accessCheck` method.